source/agora/node/Ledger.d

```diff
@@ -129,10 +129,25 @@ public class Ledger
         if (this.utxo_set.length == 0)
         {
             Block block;
+            Block last_block = gen_block;
             foreach (height; 0 .. this.last_block.header.height + 1)
             {
                 this.storage.readBlock(block, Height(height));
+
+                if (height != 0)
```

Comment:
You implemented `isGenesisBlockInvalidReason`. Why not use it here?

Reply:
It can be used now. :)

Reply:
Excellent.